tests/test_love_producer_base.py

```diff
@@ -0,0 +1,376 @@
+            self.producer.get_initial_state_as_json()
+
+    def test_get_initial_state_as_json(self):
+
+        component_name = "Test"
+
+        self.producer.component_name = component_name
+
+        initial_state_subscribe_msg = json.loads(
+            self.producer.get_initial_state_as_json()
+        )
+
+        for key, value in [
+            ("option", "subscribe"),
+            ("category", "initial_state"),
+            ("csc", component_name),
+            ("salindex", "all"),
+            ("stream", "all"),
+        ]:
+            self.assertIn(key, initial_state_subscribe_msg)
+            self.assertEqual(initial_state_subscribe_msg[key], value)
+
+    def test_add_metadata(self):
+
+        self.setup_for_data_handling_test()
+
+        self.producer.add_metadata(component_index=1)
+
+        message = self.producer.get_message_category_as_json(
+            category="telemetry", data_as_dict=dict()
+        )
+
+        data_as_dict = json.loads(message)
+
+        self.assertIn("component_index", data_as_dict)
+        self.assertEqual(data_as_dict["component_index"], 1)
+
+    async def test_register_monitor_data_periodically_with_function(self):
+
+        self.setup_for_data_handling_test()
+
+        self.producer.register_monitor_data_periodically(
+            self.get_random_data, "telemetry"
+        )
+
+        await self.assert_monitored_data("get_random_data", 2)
+
+    async def test_register_monitor_data_periodically_fail_component_not_set(self):
+        with self.assertRaises(AssertionError):
+            self.producer.register_monitor_data_periodically(
+                self.async_get_random_data, "telemetry"
+            )
+
+    async def test_register_monitor_data_periodically_with_coroutine(self):
+
+        self.setup_for_data_handling_test()
+
+        self.producer.register_monitor_data_periodically(
+            self.async_get_random_data, "telemetry"
+        )
+
+        await self.assert_monitored_data("async_get_random_data", 2)
+
+    async def test_handle_asynchronous_data_callback(self):
+
+        self.setup_for_data_handling_test()
+
+        self.setup_data_generation_loop()
+
+        self.generate_async_data_callback = (
+            self.producer.handle_asynchronous_data_callback
+        )
+
+        await self.assert_monitored_data("async_get_random_data_randomly", 3)
+
+    async def test_register_additional_action(self):
+
+        self.setup_for_data_handling_test()
+
+        self.setup_data_generation_loop()
+
+        self.producer.register_additional_action(
+            "async_get_random_data_randomly", unittest.mock.AsyncMock()
+        )
+        self.producer.register_additional_action("none", unittest.mock.AsyncMock())
+
+        self.generate_async_data_callback = (
+            self.producer.handle_asynchronous_data_callback
+        )
+
+        await self.assert_monitored_data("async_get_random_data_randomly", 3)
+
+        self.producer._additional_data_callbacks[
```

Comment:
Maybe the `_additional_data_callbacks` should have a `@getter` method?

Reply:
I reviewed this again and I think the getter method is not neccesary. I'll resolve the conversation